c/tskit/trees.h

```diff
@@ -265,15 +265,188 @@ typedef struct {
 @defgroup TREESEQ_API_GROUP Tree sequence API
 @{
 */
+
+/**
+@brief Initialises the tree sequence by (optionally) copying the supplied
+table collection, checking its integrity then building index maps.
+
+@rst
+This must be called before any operations are performed on the tree sequence.
+See the :ref:`sec_c_api_overview_structure` for details on how objects
+are initialised and freed.
+
+**Options**
+
+Options can be specified by providing one or more of the following bitwise
+flags:
+
+TSK_TAKE_TABLES
+    By default the table collection is copied, however if this flag is
+    specified the table collection will be taken and owned with a reference
+    held by this tree sequence. The table collection will be free'd with
+    this tree sequence.
+
+TSK_BUILD_INDEXES
+    If specified edge indexes will be built and stored in the table collection
```

Comment:
Is TSK_NO_CHECK_INTEGRITY an option here/

Reply:
No, integrity checking is always performed. I guess that could be added as an optimisation for those confident in their tables?

Reply:
That's fine, so long as the door is open for it.